record/record.app.js

```diff
@@ -58,6 +60,53 @@
 
         ERMrest.appLinkFn(UriUtils.appTagToURL);
 
+        function getPageSize(obj){
+            return ((!angular.isUndefined(obj) && obj.display.defaultPageSize)?obj.display.defaultPageSize:constants.defaultPageSize);
+        }
+
+        function getRelatedTableData(refObj,displaytype,accordionOpen,callback){
+
+            var pageSize = getPageSize(refObj);
+            refObj.read(pageSize).then(function (page) {
+                var model = {
+                    reference: refObj,
+                    columns: refObj.columns,
+                    page: page,
+                    pageLimit: pageSize,
+                    hasNext: page.hasNext,      // used to determine if a link should be shown
+                    hasLoaded: true,            // used to determine if the current table and next table should be rendered
+                    open: accordionOpen,                 // to define if the accordion is open or closed
+                    enableSort: true,           // allow sorting on table
+                    sortby: null,               // column name, user selected or null
+                    sortOrder: null,            // asc (default) or desc
+                    rowValues: [],              // array of rows values
+                    search: null,                // search term
+                    displayType: displaytype,//refObj.reference.display.type,
+                    context: "compact/brief",
+                    fromTuple: $rootScope.tuple
+                };
+                model.rowValues = DataUtils.getRowValuesFromPage(page);
+                model.config = {
+                    viewable: true,
+                    editable: $rootScope.modifyRecord,
+                    deletable: $rootScope.modifyRecord && $rootScope.showDeleteButton,
+                    selectable: false
+                };
+                // return model;
+                callback(model);
+        },function readFail(error) {
```

Comment:
And also please add comments on your functions or pieces of code that might be necessary.